docs/topics/run-code-snippets.md

```diff
@@ -55,30 +55,135 @@ To rebuild the module automatically before each run of a scratch or a worksheet,
+
+## Command line: ki shell
+
+[ki shell](https://github.com/Kotlin/kotlin-interactive-shell) (_Kotlin Interactive Shell_) is a command-line
+utility for running Kotlin code in the terminal. It's available for Linux, macOS, and Windows.
+
+ki shell provides basic code evaluation capabilites along with advanced features such as:
+* code completion
+* type checks
+* external dependencies
+* paste mode for code snippets
+* scripting support
+* and even plugin-based extensibility.
+
+### Installation
+
+To install ki shell, download its latest version from [GitHub](https://github.com/Kotlin/kotlin-interactive-shell) and 
+unzip it in a directory of your choice.
+
+On macOS, you can also install ki shell with Homebrew by running the following command:
+
+```Shell
+brew install ki
+```
+
+### Features 
+
+To start ki shell, run `bin/ki.sh` on Linux and macOS (or just `ki` if ki shell in installed with Homebrew) or
+`bin\ki.bat` on Windows.
+
+When the shell is started, you can write Kotlin code right away in your terminal.
+
+The shell provides code completion upon pressing **Tab**. It also provides the syntax highlighting as you type;
+you can disable it by entering `:syntax off`.
+
+![ki shell highlighting and completion](ki-shell-highlight-completion.png){width=700}
+
+When you press **Enter**, ki shell evaluates the entered line and prints the result. Expression values are
+printed as variables with auto-generated names like `res*`. You can later use such variables in the code you run. 
+If the entered construct is incomplete (for example, an `if` with a condition but without the body), the shell prints 
+three dots and expects the remaining part.
+
+
+![ki shell results](ki-shell-results.png){width=700}
+
+Type in `:help` (or `:h`) to see commands available in ki shell. Below is a quick overview of some helpful ones.
+
+#### Find out an expression's type
+
+For complex expressions or APIs you don't know well, ki shell provides the command `:type` (or `:t`) that shows the 
+type of an expression:
+
+![ki shell type](ki-shell-type.png){width=700}
+
+#### Load code
+
+If the code you need is stored somewhere outside, there are two ways to load it and use in ki shell:
+* load a source file with the `:load` (or `:l`) command
+* copy and paste the code snippet in the paste mode with the `:paste` (or `:p`) command
+
+![ki shell load file](ki-shell-load.png){width=700}
+
+The `ls` command shows available symbols (variables and functions).
+
+#### Add external dependencies
+
+Along with the standard library, ki shell supports external dependencies.
+This lets you use it for trying third-party libraries without creating a whole project. 
+
+To add a third-party library in ki shell, use the `:dependsOn` command. By default, ki shell works with Maven Central,
+but you can use other repositories if you connect them with the `:repository` command:
+
+![ki shell external dependency](ki-shell-dependency.png){width=700}
+
+### Extensibility
```

Comment:
I'm not sure if we want to advertise the plugin authoring yet. I would skip this section and refer to the ki repository for further details (without mentioning extensibility here at all).

Reply:
Thanks. Removed the part about extensibility and added the GitHub link explicitly in the intro.